ReactWindows/ReactNative.Shared.Tests/UIManager/ViewManagerExtensionTests.cs

```diff
@@ -0,0 +1,189 @@
+            Assert.IsTrue(props.ContainsKey("bar"));
+            Assert.IsTrue(props.ContainsKey("qux"));
+            Assert.AreEqual("number", props["foo"]);
+            Assert.AreEqual("String", props["bar"]);
+            Assert.AreEqual("String", props["qux"]);
+        }
+
+        [Test]
+        public void ViewManagerExtension_UpdateProperties()
+        {
+            var vmx = new TestViewManagerExtension();
+            var foo = 42;
+            var diffMap = new ReactStylesDiffMap(new JObject
+            {
+                { "foo", foo },
+            });
+
+            var count = 0;
+            vmx.SetFooAction = x =>
+            {
+                Assert.AreEqual(foo, x);
+                count++;
+            };
+
+            vmx.UpdateProperties(null, diffMap);
+            Assert.AreEqual(1, count);
+        }
+
+        [Test]
+        public void ViewManagerExtension_UpdateProperties_Group()
+        {
+            var vmx = new TestViewManagerExtension();
+            var v1 = "hello";
+            var v2 = "world";
+            var diffMap = new ReactStylesDiffMap(new JObject
+            {
+                { "bar", v1 },
+                { "qux", v2 },
+            });
+
+            var count = 0;
+            vmx.SetOtherAction = (index, x) =>
+            {
+                Assert.IsTrue(index == 0 || index == 1);
+                Assert.AreEqual(index == 0 ? v1 : v2, x);
+                count++;
+            };
+
+            vmx.UpdateProperties(null, diffMap);
+            Assert.AreEqual(2, count);
+        }
+
+        [Test]
+        public void ViewManagerExtension_OnAfterUpdateTransaction()
+        {
+            var vmx = new TestViewManagerExtension();
+            var count = 0;
+            vmx.OnAfterUpdateTransactionAction = () => count++;
+            vmx.UpdateProperties(null, new ReactStylesDiffMap(new JObject()));
+            Assert.AreEqual(1, count);
+        }
+
+        #region Test Classes
+
+        class DummyViewManager : IViewManager
+        {
+            public string Name => throw new NotImplementedException();
+
+            public Type ShadowNodeType => throw new NotImplementedException();
+
+            public IReadOnlyDictionary<string, object> CommandsMap => throw new NotImplementedException();
+
+            public IReadOnlyDictionary<string, object> ExportedCustomBubblingEventTypeConstants => throw new NotImplementedException();
+
+            public IReadOnlyDictionary<string, object> ExportedCustomDirectEventTypeConstants => throw new NotImplementedException();
+
+            public IReadOnlyDictionary<string, object> ExportedViewConstants => throw new NotImplementedException();
+
+            public IReadOnlyDictionary<string, string> NativeProperties => throw new NotImplementedException();
+
+            public ReactShadowNode CreateShadowNodeInstance()
+            {
+                throw new NotImplementedException();
```

Comment:
consistently using arrow functions as above would help make this more congruent, declarative, and compact.